fontforge/splinestroke.c

```diff
@@ -3129,11 +3129,11 @@ return( NULL );
 		int isnext=sp->next->to==sp2, isprev=sp->prev->from==sp2;
 		BreakLine(sp->next,start,end,&first1,&second1);
 		BreakLine(sp2->next,end,start,&first2,&second2);
-		if ( first1->me.x!=second2->me.x || first1->me.y!=second2->me.y ) {
+		if ( first1->me.x!=second2->me.x || first1->me.y!=second2->me.y || first1==second2 ) {
```

Comment:
This one from #2832 should do it: https://github.com/fontforge/fontforge/files/473864/j.zip or for a larger test case: https://github.com/fontforge/fontforge/files/473448/bad-icon.svg.zip